### ISSUE TITLE
QueryClient.resetQueries([...])

### ISSUE THREAD

**Issue by PetrKubes97** [KRTirtho/fl-query#52]
For my project, this would be more useful than refetchQueries, as I don't want to be loading too much stuff in the background.

**Comment by KRTirtho** [comment]
I think what you want to achieve is invalidateQueries? You want to just make them stale but don't want the queries to refresh immediately and only refresh when required?

**Comment by PetrKubes97** [comment]
Yes, exactly

**Comment by PetrKubes97** [comment]
I've solved it with an extension. Let me know if it makes sense to you to include in the library. 

```
extension FLQueryClient on QueryClient {
  void invalidateWithPrefix(String prefix) {
    final queries = getQueriesWithPrefix(prefix);
    for (var query in queries) {
      query.reset();
      if (query.hasListeners) {
        query.refresh();
      }
    }
  }
}
```

**Comment by KRTirtho** [comment]
Yeah actually it was planned already. But never made it in a release.

But what you're suggesting, has `query.refresh` so it doesn't actually make any difference that refreshWithPrefix

**Comment by PetrKubes97** [comment]
Are you sure it does not make any difference? 

I've had a lot of queries and calling refreshWithPrefix had refreshed even the inactive ones, resulting in 429 errors and unnecessary server load. 

My implementation seems to correctly mark the queries for reload only after user visits their page.
